### PR TITLE
Rename PlanBuilder's localPartition to prevent confusion

### PR DESCRIPTION
**Pull request by kewang1024**

### CODE REVIEW

File: velox/exec/tests/utils/PlanBuilder.h [hunk 1]
```diff
@@ -584,7 +584,7 @@ class PlanBuilder {
 
   /// A convenience method to add a LocalPartitionNode with a single source (the
   /// current plan node) and hive bucket property.
-  PlanBuilder& localPartition(
+  PlanBuilder& localPartitionByBucket(
```
Comment (xiaoxmeng) [inline velox/exec/tests/utils/PlanBuilder.h:587]: We don't have any place to use localPartition with bucket property? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@kewang1024 thanks for the change!

File: velox/exec/tests/utils/PlanBuilder.h (hunk 1 above)
Comment (kewang1024) [reply velox/exec/tests/utils/PlanBuilder.h:587]: What do you mean? In TableWriteTest.cpp above, we use localPartitionByBucket

Comment (xiaoxmeng) [reply velox/exec/tests/utils/PlanBuilder.h:587]: I see. Only one place use it. Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@kewang1024 thanks for taking care of this!

**Pull request closed without merging**